[ADD] purchase: add new temp rr doc (#7576)
odoo/documentation: content/applications/inventory_and_mrp/purchase/products/temporary_reordering.rst

==========================
Temporary reordering rules
==========================

Some businesses require certain products to always have a minimum quantity of stock on-hand at any
given time. To avoid stock falling below a certain threshold, companies can create *reordering
rules* in Odoo to automate purchase orders for specific products.

Reordering rules keep the forecasted stock levels above a certain threshold without exceeding a
specified upper limit. When a product has a reordering rule applied to it, and that product's
quantity falls below a specified number, Odoo uses the specified *route* on the product form
(e.g. *Buy* or *Manufacture*), and creates an order to replenish the stock.

However, there are some instances in which businesses might not want or need specific products to be
replenished automatically. In such cases, they can rely on "temporary" reordering rules.

> In Odoo, when a product has 0 quantity in stock, has a *Buy* route selected, and is included in a
> sales order (SO), Odoo creates a "temporary" reordering rule in the replenishment dashboard for that
> product, which is deleted upon confirmation of the purchase order (PO) created for that product.

Reviewer: Optional rewording suggestion to move all the conditions to trigger the reordering rule in a list since I personally found it a little difficult to keep track of them all. Feel free to reword this more!
```suggestion
In Odoo, a "temporary" reordering rule is created in the replenishment dashboard when a product:

#. with a *Buy* route
#. has no reordering rule configured
#. has 0 quantity in stock
#. is included in a sales order (SO).

This rule is deleted upon confirmation of the purchase order (PO) generated for the product.
```

.. seealso::
   - :doc:`../../inventory/product_management/product_replenishment/reordering_rules`
   - :doc:`../../purchase/products/reordering`

Configuration
=============

To trigger temporary reordering rules once its stock reaches 0, a product needs to be configured in
a specific way.

To create a product that will trigger temporary reordering rules, go to :menuselection:`Inventory
app --> Products --> Products`, and click :guilabel:`New`.

> On the product form, enter a name for the product, and enable selling and reordering by checking the
> :guilabel:`Can be Sold` and :guilabel:`Can be Purchased` options underneath the :guilabel:`Product
> Name` field. Finally, set the :guilabel:`Product Type` to `Storable Product`, under the
> :guilabel:`General Information` tab.

Reviewer: since "can be sold" and "can be purchased" are enabled by default for products created in the inventory app, let's make this part a little shorter (:
```suggestion
On the product form, enter the product name, and ensure the :guilabel:`Can be Sold` and
:guilabel:`Can be Purchased` options are enabled underneath the :guilabel:`Product Name` field.
Finally, set the :guilabel:`Product Type` to `Storable Product`, under the :guilabel:`General
Information` tab.
```

Next, click the :guilabel:`Purchase` tab. Under :guilabel:`Vendor`, click :guilabel:`Add a line`,
select a vendor from the drop-down menu, and set a purchase price under :guilabel:`Price`.

.. important::

>    A vendor **must** be set in the :guilabel:`Purchase` tab of a product, or temporary reordering
>    rules will not work. Although a :abbr:`PO (purchase order)` will still be created automatically,
>    if an attempt is made to replenish the product from the :guilabel:`Replenishment` dashboard in
>    the *Inventory* app, a warning appears prompting a vendor to be added on the product form.

Reviewer: ```suggestion
   A vendor **must** be set for temporary reordering rules to work. Although a :abbr:`PO (purchase
   order)` will still be created automatically, attempting to replenish the product from the
   :guilabel:`Replenishment` dashboard in the *Inventory* app triggers a warning to add a vendor on
   the product form.
```

then click :guilabel:`Add a product` under :guilabel:`Product`, and select the desired product from
the drop-down menu. Then, :guilabel:`Confirm` the :abbr:`SO (sales order)`.

.. image:: temporary_reordering/temporary-reordering-sales-order.png
   :align: center
   :alt: Sales order for product with no set reordering rules.

Check replenishment report
==========================

To see the temporary reordering rule created for the out-of-stock product included in the sales
order, navigate to :menuselection:`Inventory app --> Operations --> Replenishment`. This opens the
:guilabel:`Replenishment` dashboard.

On this dashboard, locate the product for which the temporary reordering rule was created. On its
product line, its :guilabel:`On Hand` quantity, negative :guilabel:`Forecast` quantity, *Buy*
:guilabel:`Route`, and :guilabel:`To Order` quantity to replenish can be seen. Additionally, two
replenishment options are located to the far-right of the row: :guilabel:`Order Once` and
:guilabel:`Automate`.

.. image:: temporary_reordering/temporary-reordering-replenishment-dashboard.png
   :align: center
   :alt: Replenishment report displaying temporary reordering rule and options.

> To use the one-time, temporary reordering rule, click :guilabel:`Order Once`. Doing so causes a
> confirmation banner to appear in the top-right corner, reading *The following replenishment order
> has been generated*, with a new purchase order number.

Reviewer: I think a guilabel is more suitable here
```suggestion
To use the one-time, temporary reordering rule, click :guilabel:`Order Once`. This action triggers a
confirmation pop-up in the top-right corner, reading :guilabel:`The following replenishment order
has been generated`, along with a new purchase order number.
```

given time. To avoid stock falling below a certain threshold, companies can create *reordering
rules* in Odoo to automate purchase orders for specific products.

Reordering rules keep the forecasted stock levels above a certain threshold without exceeding a
specified upper limit. When a product with a reordering rule falls below a specified quantity, Odoo
generates an order using the specified *route* (e.g. *Buy* or *Manufacture*) to replenish the stock.

In certain cases, businesses might opt for *temporary reordering rules* when they do not want
specific products to be replenished automatically.

In Odoo, a "temporary" reordering rule is created in the replenishment dashboard when a product:

#. with a *Buy* route
#. has no reordering rule configured
#. has 0 quantity in stock
#. is included in a sales order (SO).

This rule is deleted upon confirmation of the purchase order (PO) generated for the product.

.. seealso::
   - :doc:`../../inventory/product_management/product_replenishment/reordering_rules`
   - :doc:`../../purchase/products/reordering`

Configuration
=============

To configure a product that triggers temporary reordering rules when its stock reaches 0, begin by

> going to :menuselection:`Inventory app --> Products --> Products`, and click :guilabel:`New`.

Author: Yes, added a note to indicate that

On the product form, enter the product name, and ensure the :guilabel:`Can be Sold` and
:guilabel:`Can be Purchased` options are enabled underneath the :guilabel:`Product Name` field.
Finally, set the :guilabel:`Product Type` to `Storable Product`, under the :guilabel:`General
Information` tab.

Next, click the :guilabel:`Purchase` tab. Under :guilabel:`Vendor`, click :guilabel:`Add a line`,

> select a vendor from the drop-down menu, and set a purchase price under :guilabel:`Price`.

Reviewer 2: In the 17.0 runbot I used for this review, the column was titled, "Supplier" not "Vendor" -- might be worth looking into to ensure the correct title is used.

> select a vendor from the drop-down menu, and set a purchase price under :guilabel:`Price`.

Author: It's "Vendor" for me in a bunch of different runbots, I've never seen it say supplier

.. important::

>    A vendor **must** be set for temporary reordering rules to work. Although a :abbr:`PO (purchase
>    order)` will still be created automatically, attempting to replenish the product from the
>    :guilabel:`Replenishment` dashboard in the *Inventory* app triggers a warning to add a vendor on
>    the product form.

Reviewer 2: ```suggestion
   A vendor **must** be set on the product form for temporary reordering rules to work.
   
   While a :abbr:`PO (purchase order)` can still be created automatically, attempting to replenish
   the product from the :guilabel:`Replenishment` dashboard in the *Inventory* app triggers a
   warning to add a vendor on the product form.
```

   .. image:: temporary_reordering/temporary-reordering-warning-popup.png
      :align: center
      :alt: Warning pop-up upon clicking to replenish product with no set vendor.

> Before creating a :abbr:`SO (sales order)` for the product, ensure that the :guilabel:`On Hand`
> smart button on the product form reads `0.00 Units`. Then, ensure that the :guilabel:`Reordering
> Rules` smart button reads `0`, meaning there are no rules applied to this product.

Reviewer 2: ```suggestion
Before creating a :abbr:`SO (sales order)` for the product, ensure the :guilabel:`On Hand` smart
button on the product form reads `0.00 Units`. Then, ensure the :guilabel:`Reordering Rules` smart
button reads `0`, indicating there are no rules applied to this product.
```

.. image:: temporary_reordering/temporary-reordering-smart-buttons.png
   :align: center
   :alt: Product form smart button row displaying reordering rules and on hand buttons.

Create sales order

> ==================

Reviewer 2: May want to consider re-titling this: "Trigger temporary reordering rule" (or something like that)

To trigger a temporary reordering rule, create a new sales order for the previously-created product.

Navigate to :menuselection:`Sales app --> New`. Add a customer in the :guilabel:`Customer` field,
then click :guilabel:`Add a product` under :guilabel:`Product`, and select the desired product from

> the drop-down menu. Then, :guilabel:`Confirm` the :abbr:`SO (sales order)`.

Author: Can be any one configured correctly, clarified that here

.. image:: temporary_reordering/temporary-reordering-sales-order.png
   :align: center
   :alt: Sales order for product with no set reordering rules.

Check replenishment report
==========================

> To see the temporary reordering rule created for the out-of-stock product included in the sales
> order, navigate to :menuselection:`Inventory app --> Operations --> Replenishment`. This opens the
> :guilabel:`Replenishment` dashboard.

Reviewer 2: ```suggestion
To see the temporary reordering rule created for the out-of-stock product included in a sales order,
navigate to :menuselection:`Inventory app --> Operations --> Replenishment`. Doing so opens the
:guilabel:`Replenishment` dashboard.
```

> On this dashboard, locate the product for which the temporary reordering rule was created. On its
> product line, its :guilabel:`On Hand` quantity, negative :guilabel:`Forecast` quantity, *Buy*
> :guilabel:`Route`, and :guilabel:`To Order` quantity to replenish can be seen. Additionally, two
> replenishment options are located to the far-right of the row: :guilabel:`Order Once` and
> :guilabel:`Automate`.

Reviewer 2: ```suggestion
On this dashboard, locate the product for which the temporary reordering rule was created. On its
product line, its :guilabel:`On Hand` quantity, negative :guilabel:`Forecast` quantity, *Buy*
:guilabel:`Route`, and :guilabel:`To Order` quantity to replenish can be seen.

Additionally, two replenishment options are located to the far-right of the row: :guilabel:`Order
Once` and :guilabel:`Automate`.
```

To use the one-time, temporary reordering rule, click :guilabel:`Order Once`. This action triggers a
confirmation pop-up in the top-right corner, reading :guilabel:`The following replenishment order
has been generated`, along with a new purchase order number.

.. tip::
   Once the purchase order has been generated after clicking :guilabel:`Order Once`, refresh the
   page. The temporary reordering rule for the product no longer appears in the
   :guilabel:`Replenishment` dashboard.

Complete purchase order
=======================

To view the purchase order created from the :guilabel:`Replenishment` dashboard, navigate to the
:menuselection:`Purchase app`, and select the generated :abbr:`PO (purchase order)` from the
:guilabel:`Requests for Quotation` overview.

Then, click :guilabel:`Confirm Order`, click :guilabel:`Receive Products`, and finally, click
:guilabel:`Validate` to complete the purchase order.

.. image:: temporary_reordering/temporary-reordering-purchase-order.png
   :align: center
   :alt: Purchase order for product ordered with temporary reordering rule.

Now, the original sales order can be delivered and invoiced.

.. note::

>    Once the :abbr:`SO (sales order)` is delivered and invoiced, check that there are no reordering
>    rules on the product form.

Reviewer 2: ```suggestion
   Once the :abbr:`SO (sales order)` is delivered and invoiced, ensure there are no reordering rules
   on the product form.
```

>    Go to :menuselection:`Inventory app --> Products --> Products`, and select the previously-created
>    product. Confirm that the :guilabel:`Reordering Rules` smart button displays `0`.

Reviewer 2: ```suggestion
   Go to :menuselection:`Inventory app --> Products --> Products`, select the product, and confirm
   that the :guilabel:`Reordering Rules` smart button displays `0`.
```

==========================
Temporary reordering rules
==========================

Some businesses require certain products to always have a minimum quantity of stock on-hand at any
given time. To avoid stock falling below a certain threshold, companies can create *reordering
rules* in Odoo to automate purchase orders for specific products.

Reordering rules keep the forecasted stock levels above a certain threshold, without exceeding a
specified upper limit, or maximum amount. When a product with a reordering rule falls below a
specified quantity, Odoo generates an order using the specified *route* (e.g. *Buy* or
*Manufacture*) to replenish the stock.

In certain cases, businesses might opt for *temporary reordering rules* when they do not want
specific products to be replenished automatically.

In Odoo, a "temporary" reordering rule is created in the replenishment dashboard when a product:

> #. with a *Buy* route

Reviewer 2: ```suggestion
#. is configured with a *Buy* route
```

.. image:: temporary_reordering/temporary-reordering-sales-order.png
   :align: center
   :alt: Sales order for product with no set reordering rules.

Check replenishment report
==========================

To see the temporary reordering rule created for the out-of-stock product included in the sales
order, navigate to :menuselection:`Inventory app --> Operations --> Replenishment`. Doing so opens
the :guilabel:`Replenishment` dashboard.

On this dashboard, locate the product for which the temporary reordering rule was created. On its
product line, its :guilabel:`On Hand` quantity, negative :guilabel:`Forecast` quantity, *Buy*
:guilabel:`Route`, and :guilabel:`To Order` quantity to replenish can be seen.

Additionally, two replenishment options are located to the far-right of the row: :guilabel:`Order
Once` and :guilabel:`Automate`.

.. image:: temporary_reordering/temporary-reordering-replenishment-dashboard.png
   :align: center
   :alt: Replenishment report displaying temporary reordering rule and options.

To use the one-time, temporary reordering rule, click :guilabel:`Order Once`. This action triggers a

> confirmation pop-up in the top-right corner, reading :guilabel:`The following replenishment order
> has been generated`, along with a new purchase order number.

Reviewer 2: ```suggestion
confirmation pop-up window in the top-right corner, reading :guilabel:`The following replenishment
order has been generated`, along with a new purchase order number.
```

.. tip::
   Once the purchase order has been generated after clicking :guilabel:`Order Once`, refresh the
   page. The temporary reordering rule for the product no longer appears in the
   :guilabel:`Replenishment` dashboard.

Complete purchase order
=======================

To view the purchase order created from the :guilabel:`Replenishment` dashboard, navigate to the
:menuselection:`Purchase app`, and select the generated :abbr:`PO (purchase order)` from the
:guilabel:`Requests for Quotation` overview.

> Then, click :guilabel:`Confirm Order`, click :guilabel:`Receive Products`, and finally, click
> :guilabel:`Validate` to complete the purchase order.

Reviewer 2: ```suggestion
From here, click :guilabel:`Confirm Order`, then click :guilabel:`Receive Products`. Finally, click
:guilabel:`Validate` to complete the purchase order.
```
